rust/tools/file_header.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use anyhow::anyhow;
+use clap::Parser as _;
+use file_header::{check_headers_recursively, add_headers_recursively, license::spdx::{
+    APACHE_2_0, YearCopyrightOwnerValue
+}};
+use globset::{Glob, GlobSet, GlobSetBuilder};
+use std::{env, path::PathBuf};
+
+fn main() -> anyhow::Result<()> {
+    let rust_dir = PathBuf::from(env::var("CARGO_MANIFEST_DIR")?);
+    let ignore_globset = ignore_globset()?;
+    // Note: when adding headers, there is a bug where the line spacing is off for Apache 2.0 (see https://github.com/spdx/license-list-XML/issues/2127)
+    let header = APACHE_2_0.build_header(YearCopyrightOwnerValue::new(2023, "Google LLC".into()));
```

Comment:
will that not conflict with headers that have a date range for the copyright? (like `Copyright 2021-2022 Google LLC`)?

Reply:
The search pattern used to check for an existing license is just the date-free ASL2 header, so it shouldn't care about date ranges: https://github.com/google/file-header/blob/main/src/license/spdx/mod.rs#L327.
However, that does raise the question of perhaps we should be using the current date's year when _adding_ licenses 🤔

Reply:
Ok, so for now we can probably keep as-is.